### ISSUE TITLE
Quarkus multimodule project with jandex test faling with AmbiguousResolutionException

### ISSUE THREAD

**Issue by guitcastro** [quarkusio/quarkus#32920]
### Describe the bug

I have a multi-module project setup (`app` and `infrastructure`),  the `infrastructure` uses jandex to have the beans available on parent(`app`) module. However when I try to run  `infrastructure` tests I got the following error:

```
javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type com.simpleps.domain.quarkustemplate.SampleRepository and qualifiers [@Default]
        - java member: com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest#sampleRepository
        - declared on CLASS bean [types=[java.lang.Object, com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest], qualifiers=[@Default, @Any], target=com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest]

```

### Expected behavior

The tests should be working without AmbiguousResolutionException

### Actual behavior

The tests fail with AmbiguousResolutionException

### How to Reproduce?

Reproduces:

1. Clone the repository: https://github.com/guitcastro/quarkus-jandex-bug
2. run `./gradlew infrastructure:build` and see that the test will fail
3. run ./gradlew app:build` and the test will pass
4. comment out `id ("org.kordamp.gradle.jandex") version "1.1.0"` on `infrastructure` build.kts 
5. run `./gradlew infrastructure:build` and see that the test will pass
6. run ./gradlew app:build` and the test will fail because of a missing bean

### Output of `uname -a` or `ver`

Darwin MacBook-Pro-de-Guilherme.local 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 LTS OpenJDK Runtime Environment GraalVM 22.3.0 (build 17.0.5+8-LTS) OpenJDK 64-Bit Server VM GraalVM 22.3.0 (build 17.0.5+8-LTS, mixed mode, sharing)

### GraalVM version (if different from Java)

openjdk version "17.0.5" 2022-10-18 LTS OpenJDK Runtime Environment GraalVM 22.3.0 (build 17.0.5+8-LTS) OpenJDK 64-Bit Server VM GraalVM 22.3.0 (build 17.0.5+8-LTS, mixed mode, sharing)

### Quarkus version or git rev

2.16.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle 8.1.1

**Comment by geoand** [comment]
cc @mkouba @Ladicek

**Comment by mkouba** [comment]
For the record, the full error message is:
```
  javax.enterprise.inject.AmbiguousResolutionException: Ambiguous dependencies for type com.simpleps.domain.quarkustemplate.SampleRepository and qualifiers [@Default]
                        - java member: com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest#sampleRepository
                        - declared on CLASS bean [types=[java.lang.Object, com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest], qualifiers=[@Default, @Any], target=com.simpleps.infrastructure.quarkustemplate.SampleRepositoryTest]
                        - available beans:
                                - CLASS bean [types=[com.simpleps.infrastructure.quarkustemplate.SampleRepositoryImpl, com.simpleps.domain.quarkustemplate.SampleRepository, java.lang.Object], qualifiers=[@Default, @Any], target=com.simpleps.infrastructure.quarkustemplate.SampleRepositoryImpl]
                                - CLASS bean [types=[com.simpleps.infrastructure.quarkustemplate.SampleRepositoryImpl, com.simpleps.domain.quarkustemplate.SampleRepository, java.lang.Object], qualifiers=[@Default, @Any], target=com.simpleps.infrastructure.quarkustemplate.SampleRepositoryImpl]
```

Quarkus also detected a split package:
```
    WARN: Detected a split package usage which is considered a bad practice and should be avoided. Following packages were detected in multiple archives: 
    - "com.simpleps.infrastructure.quarkustemplate" found in [d5c6fed0e590209e1aa279325ce2de3f3d9faa48:infrastructure::jar, quarkustemplate:infrastructure::jar]
```

> comment out id ("org.kordamp.gradle.jandex") version "1.1.0" on infrastructure build.kts

What is this library used for? To make a jandex index in the `infrastructure` module so that the beans are discovered? If so then you could try to add a [`beans.xml` descriptor](https://quarkus.io/guides/cdi-reference#bean_discovery) instead.

**Comment by Ladicek** [comment]
Looks like the same JAR is present in the application _twice_?

**Comment by mkouba** [comment]
CC @aloubyansky

**Comment by mkouba** [comment]
>     4. comment out `id ("org.kordamp.gradle.jandex") version "1.1.0"` on `infrastructure` build.kts
> 
>     5. run `./gradlew infrastructure:build` and see that the test will pass
> 
>     6. run ./gradlew app:build` and the test will fail because of a missing bean

I can confirm this observation. And if I remove the `jandex` plugin and add an empty `beans.xml`  then I get the `AmbiguousResolutionException` again.

**Comment by mkouba** [comment]
> The sample project that was gracefully provided, successfully builds and runs the tests, when downgrading Kotlin's version 1.8.10.

So it seems to be related to the kotlin version used...

**Comment by andrezimmermann** [comment]
I'm very sorry, I had to make some changes to the project to actually compile and run the tests but didn't occurred me to share those changes.

I've created a fork on the original repository. For the Kotlin [1.7.22](https://github.com/andrezimmermann/quarkus-jandex-bug/tree/kotlin/1.7.22) and [1.8.20](https://github.com/andrezimmermann/quarkus-jandex-bug/tree/kotlin/1.8.20)

I'm hoping that would be simple cloning these branches as running `./gradlew build` to be able reproducing the error.
Please let me know if you have any issues with it

**Comment by geoand** [comment]
Not sure what's wrong with my setup, but I am getting the same error I mentioned above, so I won't be able to investigate this.

**Comment by andrezimmermann** [comment]
That is fair. I've created my own reproducer for the issue I'm facing and opened a new ticket with my findings.
https://github.com/quarkusio/quarkus/issues/34061

**Comment by guitcastro** [comment]
This should be closed, since #34061  states that the bug is related to kotlin version and fixed in kotlin 1.8.22
